posthog/session_recordings/queries/test/listing_recordings/test_session_recordings_list_by_cohort.py

```diff
@@ -0,0 +1,511 @@
+                    session_id=session_id_one,
+                    first_timestamp=self.an_hour_ago + relativedelta(seconds=30),
+                    team_id=self.team.id,
+                )
+                produce_replay_summary(
+                    distinct_id=user_two,
+                    session_id=session_id_two,
+                    first_timestamp=self.an_hour_ago,
+                    team_id=self.team.id,
+                )
+                create_event(
+                    user_two,
+                    self.an_hour_ago,
+                    team=self.team,
+                    event_name="custom_event",
+                    properties={"$session_id": session_id_two},
+                )
+                produce_replay_summary(
+                    distinct_id=user_two,
+                    session_id=session_id_two,
+                    first_timestamp=self.an_hour_ago + relativedelta(seconds=30),
+                    team_id=self.team.id,
+                )
+
+                self._assert_query_matches_session_ids(
+                    {
+                        # has to be in the cohort and pageview has to be in the events
+                        # test data has one user in the cohort but no pageviews
+                        "properties": [
+                            {
+                                "key": "id",
+                                "value": cohort.pk,
+                                "operator": "in",
+                                "type": "cohort",
+                            }
+                        ],
+                        "events": [
+                            {
+                                "id": "$pageview",
+                                "type": "events",
+                                "order": 0,
+                                "name": "$pageview",
+                            }
+                        ],
+                    },
+                    [],
+                )
+
+                self._assert_query_matches_session_ids(
+                    {
+                        "properties": [
+                            {
+                                "key": "id",
+                                "value": cohort.pk,
+                                "operator": "in",
+                                "type": "cohort",
+                            }
+                        ],
+                        "events": [
+                            {
+                                "id": "custom_event",
+                                "type": "events",
+                                "order": 0,
+                                "name": "custom_event",
+                            }
+                        ],
+                    },
+                    [session_id_two],
+                )
+
+    @snapshot_clickhouse_queries
+    @also_test_with_materialized_columns(person_properties=["$some_prop"])
+    def test_internal_account_filter_with_cohort_properties(self) -> None:
```

Comment:
everything else is copied out of the main file...
this is the new test
there's a bunch of duplication here but this PR is already pretty huge and these listing tests are already very duplicatey
and want to get this out and verify the fix before i move things around too much more